LucidTestKit/Utils/AsyncExpectation.swift

```diff
@@ -0,0 +1,192 @@
+//
+//  Async.swift
+//  Lucid
+//
+//  Created by Ezequiel Munz on 01/12/2023.
+//  Copyright © 2023 Scribd. All rights reserved.
+//
+
+import XCTest
+
+extension XCTestCase {
+    ///
+    /// Asynchronous assertion function that allows a future expectation to be checked.
+    /// The expression passed by argument will be checked every 10 miliseconds until it gets satisfied or the timeout is hit
+    /// If the expression is eventually satisfied, the test will pass
+    /// If the timeout is hit before the expression is satisfied, the test will fail
+    ///
+    /// This function is useful to test asynchronous code where the execution happens within multiple threads
+    ///
+    /// - parameters:
+    ///     - expression: The expression to be eventually satisfied
+    ///     - timeout: The timeout interval for the operation to finish
+    ///     - file: The file where the function is being called from (We don't need to overwrite it's value)
+    ///     - line: The line where the function is being called from (We don't need to overwrite it's value)
+    ///
+    public func AsyncExpectation(expression: @escaping @autoclosure () async -> Bool, timeout: TimeInterval, file: StaticString = #file, line: UInt = #line) async {
+        await withTaskGroup(of: Void.self) { group in
+            // Timeout
+            group.addTask {
+                let timeoutValue = UInt64(1000000000 * timeout)
+                var elapsedTime: TimeInterval = Date().timeIntervalSince1970
+                try? await Task.sleep(nanoseconds: timeoutValue)
+                guard Task.isCancelled == false else { return }
+                elapsedTime = Date().timeIntervalSince1970 - elapsedTime
+                XCTFail("Timeout after: \(elapsedTime)", file: file, line: line)
+            }
+            // Assertion
+            group.addTask {
+                var success = false
+                var elapsedTime: TimeInterval = Date().timeIntervalSince1970
+                repeat {
+                    guard Task.isCancelled == false else { return }
+                    if await expression() {
+                        success = true
+                        elapsedTime = Date().timeIntervalSince1970 - elapsedTime
+                        XCTAssertTrue(success, "Comparison succeeded after \(elapsedTime) seconds", file: file, line: line)
+                        return
+                    }
+                    try? await Task.sleep(nanoseconds: NSEC_PER_MSEC * 10)
```

Comment:
@StephaneMagne We should probably review this timing since we can't convert all the Doubles to `actor` due to some compiler complains. 10 milliseconds may be good to reduce the tests total execution time, but it also may be too frequent and increase the probability of race conditions. So far I ran the tests a 100 times and didn't hit any race condition. With the `actor` doubles this is not an issue at all

Reply:
Is there a race condition with this time and `actor` objects? I thought the race condition was only with `classes`.